compiler/src/iree/compiler/Dialect/VM/Conversion/MemRefToVM/ConvertMemRefToVM.cpp

```diff
@@ -118,6 +118,36 @@ class ConvertMemRefGetGlobalOp
   }
 };
 
+class ConvertMemRefAllocaOp : public OpConversionPattern<memref::AllocaOp> {
+ public:
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      memref::AllocaOp allocaOp, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    auto type = allocaOp.getType().cast<ShapedType>();
+    if (!type.hasStaticShape()) {
+      return rewriter.notifyMatchFailure(
+          allocaOp, "unable to create buffers for dynamic shapes");
+    }
+
+    int64_t length = type.getSizeInBits();
+    if (adaptor.alignment()) {
```

Comment:
alignment specifies the base pointer alignment and not the length, so you don't need to do this here.
the vm op doesn't currently allow specifying an alignment but it probably should; filed #9165 to track that separately.

Reply:
I removed it and added a comment.